native/core/src/parquet/objectstore/s3.rs

```diff
@@ -221,14 +221,31 @@ fn normalize_endpoint(
         endpoint.to_string()
     };
 
-    if virtual_hosted_style_request {
+    if force_path_style {
+        // Path-style: bucket in path
+        // Example: https://s3.oss-me-central-1.aliyuncs.com/my-bucket/key
         if endpoint.ends_with("/") {
             Some(format!("{endpoint}{bucket}"))
         } else {
             Some(format!("{endpoint}/{bucket}"))
         }
     } else {
-        Some(endpoint) // Avoid extra to_string() call since endpoint is already a String
+        // Virtual-hosted-style: bucket in subdomain
+        // For custom endpoints (non-AWS), we need to manually insert the bucket as a subdomain
+        // because object_store crate only does this automatically for AWS S3 endpoints.
+        // Example: https://my-bucket.s3.oss-me-central-1.aliyuncs.com/key
+        
+        // Parse the endpoint to insert bucket as subdomain
+        if let Some(protocol_end) = endpoint.find("://") {
+            let protocol = &endpoint[..protocol_end + 3]; // "https://" or "http://"
+            let rest = &endpoint[protocol_end + 3..];     // "s3.oss-me-central-1.aliyuncs.com" or "s3.oss-me-central-1.aliyuncs.com/path"
+            
+            // Insert bucket as first subdomain
+            Some(format!("{protocol}{bucket}.{rest}"))
```

Comment:
According to https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html#general-purpose-bucket-names 
```
Bucket names can consist only of lowercase letters, numbers, periods (.), and hyphens (-).
```
usage of periods (.) in the bucket name would create several subdomains. Is that supported ? Or a check should be added and reported somehow ?!